### PR TITLE
combine PR tests in one workflow

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #1604

See [the comment below](https://github.com/mdn/content/pull/589#issuecomment-749805562) which summarizes it pretty well.

### CODE REVIEW

**Comment by peterbe** [comment]
@bershanskiy I ran out of time on this and the holidays are coming but it's looking pretty promising. What it does is that there's one big workflow that checks:

* Changes to `index.html` files
* Changes to `files/**/*.(png|jpeg|etc)`
* Changes to `files/**/_redirects.txt`

Before each one of those, there's a `technote-space/get-diff-action` which has a different `PATTERNS:` each time and sets a different `SET_ENV_NAME` and then. 
Each step that actually tests something is dependent on each equivalent `get-diff-action`.

Now...,

* the checkout is done only once
* The whole `yarn install` stuff is only done once and each time it potentially benefits from the whole `node_modules` being cached. 
* There's less overheads because we're only creating 1 single container for all 3 different kinds of tests.
* It doesn't check out `mdn/yari` just to get to some CLIs inside it.

At the time doing all 3 different kinds of PR tests takes a total of 36 seconds.
I love the look of this column now:
Screen Shot 2020-12-22 at 5 17 42 PM
That's not bad!
I don't know how that compares to what it *used* to take. Something to experiment with after the holidays.

**Comment by peterbe** [comment]
Ok. So, in [this PR](https://github.com/mdn/content/pull/958) it spanned edits to the `index.html` and to images. So it [runs two checks](https://github.com/mdn/content/pull/958/checks):

* [build content](https://github.com/mdn/content/pull/958/checks?check_run_id=1650598529) - 1m 7s
* [file check](https://github.com/mdn/content/pull/958/checks?check_run_id=1650598532)  - 2min 18s

So, in a sense the CI checking stuff took 2min 18s because that's the longest of two that run in parallel. 

As of *this* PR, that now becomes 36 seconds instead (on a warm `node_modules` cache). It's primarily because it can cache the whole `node_modules` which is faster than anything we did before. 

Also, what this PR introduces is that we *reuse* the `git checkout` and the `yarn install` for potentially 3 different distinct things. Yes, it's counter to time-optimization of parallelism, but it's more energy efficient because you can do it all in *1* serial workflow. 

I think this PR is ready for review now.

**Comment by peterbe** [comment]
@fiji-flo ping

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -97,4 +95,33 @@ jobs:
...
+
+      - name: Check changed files
+        if: ${{ env.GIT_DIFF_FILES }}
+        run: |
+          echo ${{ env.GIT_DIFF_FILES }}
+
+          export CONTENT_ROOT=$(pwd)/files
+          # node node_modules/@mdn/yari/filecheck/cli.js ${{ env.GIT_DIFF_FILES }}
+          yarn filecheck ${{ env.GIT_DIFF_FILES }}
+          # yarn filecheck ${{ env.GIT_DIFF_FILES }}
```
Comment (fiji-flo) [inline .github/workflows/pr-test.yml:115]:
```suggestion
```

File: .github/workflows/pr-test.yml [hunk 2]
```diff
@@ -97,4 +95,33 @@ jobs:
...
+          ABSOLUTE: true
+          SET_ENV_NAME: GIT_DIFF_FILES
+
+      - name: Check changed files
+        if: ${{ env.GIT_DIFF_FILES }}
+        run: |
+          echo ${{ env.GIT_DIFF_FILES }}
+
+          export CONTENT_ROOT=$(pwd)/files
+          # node node_modules/@mdn/yari/filecheck/cli.js ${{ env.GIT_DIFF_FILES }}
```
Comment (fiji-flo) [inline .github/workflows/pr-test.yml:113]:
```suggestion
```

File: .github/workflows/pr-test.yml [hunk 3]
```diff
@@ -97,4 +95,33 @@ jobs:
           # you don't need that script as a writer. It's only used in CI
           # and it can't use the default CONTENT_ROOT that gets set in
           # package.json.
-          node node_modules/@mdn/yari/build/cli.js ${{ env.GIT_DIFF }}
+          # node node_modules/@mdn/yari/build/cli.js ${{ env.GIT_DIFF_CONTENT }}
```
Comment (fiji-flo) [inline .github/workflows/pr-test.yml:98]:
```suggestion
```

**Review by fiji-flo — APPROVED** [review]
May remove the commented commands? Otherwise: nice cleanup.

**Pull request merged**